src/DataTypes/Serializations/SerializationBool.cpp

```diff
@@ -6,164 +6,291 @@
+{
+    serializeSimple(column, row_num, ostr, settings);
+}
+
+void SerializationBool::serializeSimple(const IColumn & column, size_t row_num, WriteBuffer & ostr, const FormatSettings &) const
+{
+    const auto * col = checkAndGetSerializeColumnType(column);
+
+    if (col->getData()[row_num])
+        ostr.write(str_true, sizeof(str_true) - 1);
+    else
+        ostr.write(str_false, sizeof(str_false) - 1);
+}
+
+void SerializationBool::deserializeWithCustom(
+    IColumn & column, ReadBuffer & istr, const FormatSettings & settings, std::function<bool(ReadBuffer &)> check_end_of_value) const
+{
+    ColumnUInt8 * col = checkAndGetDeserializeColumnType(column);
+
+    PeekableReadBuffer buf(istr);
+    buf.setCheckpoint();
+    if (checkString(settings.bool_true_representation, buf) && check_end_of_value(buf))
     {
         col->insert(true);
+        return;
     }
-    else if (settings.bool_false_representation == input)
+
+    buf.rollbackToCheckpoint();
+    if (checkString(settings.bool_false_representation, buf) && check_end_of_value(buf))
     {
         col->insert(false);
+        buf.dropCheckpoint();
+        if (buf.hasUnreadData())
+            throw Exception(
+                ErrorCodes::CANNOT_PARSE_BOOL,
+                "Cannot continue parsing after parsed bool value because it will result in the loss of some data. It may happen if "
+                "bool_true_representation or bool_false_representation contains some delimiters of input format");
+        return;
+    }
+
+    buf.rollbackToCheckpoint();
+    if (deserializeImpl(col, buf) && check_end_of_value(buf))
+    {
+        buf.dropCheckpoint();
+        if (buf.hasUnreadData())
+            throw Exception(
+                ErrorCodes::CANNOT_PARSE_BOOL,
+                "Cannot continue parsing after parsed bool value because it will result in the loss of some data. It may happen if "
+                "bool_true_representation or bool_false_representation contains some delimiters of input format");
+        return;
+    }
+
+    throw Exception(
+        ErrorCodes::CANNOT_PARSE_BOOL,
+        "Invalid boolean value, should be '{}' or '{}' controlled by setting bool_true_representation and "
+        "bool_false_representation or one of "
+        "True/False/T/F/Y/N/Yes/No/On/Off/Enable/Disable/Enabled/Disabled/1/0",
+        settings.bool_true_representation, settings.bool_false_representation);
+}
+
+bool SerializationBool::deserializeImpl(ColumnUInt8 * column, ReadBuffer & istr) const
+{
+    if (checkCharCaseInsensitive('1', istr))
+    {
+        column->insert(true);
+    }
+    else if (checkCharCaseInsensitive('0', istr))
+    {
+        column->insert(false);
+    }
+    /// 'True' and 'T'
+    else if (checkCharCaseInsensitive('t', istr))
+    {
+        /// Check if it's just short form `T` or full form `True`
+        if (checkCharCaseInsensitive('r', istr))
+        {
+            if (!checkStringCaseInsensitive("ue", istr))
```

Comment:
Typos check failes because of this `"ue"`, can we skip this line in stype-check somehow?

Reply:
Yes, we should edit style check accordingly...